### PR TITLE
chore(plugins): return plugin activated state from cache

### LINKED ISSUE #11821: elgg_is_active_plugin should use active plugins cache
currently it does a query per plugin that is questioned (using elgg_check_relationship). This can be done a lot smarter

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #11821

### CODE REVIEW

File: engine/classes/Elgg/Database/Plugins.php [hunk 1]
```diff
@@ -154,14 +154,19 @@ public function getPath() {
...
-					continue;
-				}
-				$this->cache->save($plugin->getID(), $plugin);
+		if (!is_array($plugins)) {
+			unset($this->boot_plugins);
+			return;
+		}
+		
+		foreach ($plugins as $plugin) {
+			$plugin_id = $plugin->getID();
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/Plugins.php]: This will fatal if it's not a plugin instance

File: engine/classes/ElggPlugin.php [hunk 2]
```diff
@@ -539,17 +544,12 @@ public function isValid() {
...
-		$site = elgg_get_site_entity();
-
-		if (!($site instanceof \ElggSite)) {
-			return false;
+		if (isset($this->activated)) {
+			return $this->activated;
 		}
 
-		return check_entity_relationship($this->guid, 'active_plugin', $site->guid) instanceof ElggRelationship;
+		$this->activated = elgg_is_active_plugin($this->getID());
```
Comment (hypeJunction) [inline engine/classes/ElggPlugin.php]: I don't like this. I'd prefer not to use global APIs within object instances. It should be the other way around. `elgg_is_active_plugin()` should load plugin entity and check `ElggPlugin::isActive`

File: engine/classes/Elgg/Database/Plugins.php (hunk 1 above)
Comment (jdalsem) [reply engine/classes/Elgg/Database/Plugins.php]: check

File: engine/classes/ElggPlugin.php (hunk 2 above)
Comment (jdalsem) [reply engine/classes/ElggPlugin.php]: tried it... can't figure out a good way... you can give it a try if you think there is a way

Comment (jdalsem) [reply engine/classes/ElggPlugin.php]: only the service nows activated vs inactivated plugins... letting the plugin figure it out will cost unnecessary queries

Comment (hypeJunction) [reply engine/classes/ElggPlugin.php]: Not a priority at the moment. Could you add a todo somewhere?

Comment (jdalsem) [reply engine/classes/ElggPlugin.php]: todo is here https://github.com/Elgg/Elgg/issues/11827

**Pull request merged**